static/js/composebox_typeahead.js

```diff
@@ -1092,7 +1092,7 @@ export function initialize() {
     $("form#send_message_form").on("keyup", handle_keyup);
 
     $("#enter_sends").on("click", () => {
-        page_params.enter_sends = $("#enter_sends").is(":checked");
+        page_params.user_settings.enter_sends = $("#enter_sends").is(":checked");
```

Comment:
Reading this, I wonder if we want to create a `user_settings` module to hold these settings rather than accessing them through `page_params`; we can just make `ui_init.js` initialize the `user_settings` module with the `page_prams.user_settings` object and then everything can import and access that module.  It'd certainly make code like this read more cleanly, and I think would help a lot in reducing the amount of random code that needs to interact with `page_params` at all.

Making a `user_settings` module, which would be just a dictionary with all the settings in it, like present `page_params` module, might be a pretty clean preparatory refactor we do on master, if you think it's worth doing.

Reply:
Yeah I agree. We could do it as a preparatory PR, will be easy to review this PR I think. 
We should add a typescript module, right?